src/middlewared/middlewared/plugins/enclosure.py

```diff
@@ -478,19 +504,27 @@ def map_disks_to_enclosure_slots(self):
             return
 
         disk_raw_values = dict()
-        for k, v in EnclosureDevice(f'/dev/{self.devname}').status()['elements'].items():
-            if v['type'] == 23 and v['descriptor'] != '<empty>':
-                disk_raw_values[k] = v['status']
+        if not is_hseries:
+            for k, v in EnclosureDevice(f'/dev/{self.devname}').status()['elements'].items():
+                if v['type'] == 23 and v['descriptor'] != '<empty>':
+                    disk_raw_values[k] = v['status']
 
         for slot in sorted(mapping):
-            self.append(self._enclosure_element(
-                slot,
-                'Array Device Slot',
-                self._parse_raw_value(disk_raw_values.get(slot, 0)),
-                None,
-                '',
-                mapping[slot],  # disk
-            ))
+            disk, slot_status, ident, fault = mapping[slot]
+            if is_hseries:
+                info = self._enclosure_element(slot, 'Array Device Slot', slot_status, None, '', disk, ident, fault)
+            else:
+                info = self._enclosure_element(
+                    slot,
+                    'Array Device Slot',
+                    self._parse_raw_value(disk_raw_values.get(slot, [5, 0, 0, 0])),
+                    None,
+                    '',
+                    disk
+                )
```

Comment:
Possibly worthwhile to comment **why** we didn't supply the `ident` and `fault` here too, or instead **why** we did include them for H-series.